### PR TITLE
Tabular: Fix GPU specification in FastAI to respect num_gpus

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

- Fix GPU specification in FastAI to respect num_gpus
- Previously if GPU was present, it would always be used regardless of num_gpus setting
- Fixed edge case inefficiency in ensemble_selection when identical models caused selection to choose all of them due to rounding error instead of choosing just a single one.
- Optimized ensemble selection to prioritize models already in ensemble when ties occur.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1421-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1421/1/index.html

**Comment by szha** [comment]
Job PR-1421-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1421/2/index.html

File: tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py [hunk 1]
```diff
@@ -184,18 +183,12 @@ def _fit(self,
...
 
-        if num_cpus is None:
-            num_cpus = defaults.cpus
-        # additional workers are helping only when fork is enabled; in other mp modes, communication overhead reduces performance
-        num_workers = int(num_cpus / 2)
-        if not is_fork_enabled():
-            num_workers = 0
         if num_gpus is not None:
             if num_gpus == 0:
                 # TODO: Does not obviously impact inference speed
```
Comment (gradientsky) [inline tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py:188]: TODO not needed
add TODO to respect `CUDA_VISIBLE_DEVICES` to select proper GPU

**Review by gradientsky — APPROVED** [review]

File: tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py (hunk 1 above)
Comment (Innixma) [reply tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py]: added

**Pull request merged**